web/src/add_subscribers_pill.ts

```diff
@@ -251,8 +263,19 @@ export function set_up_handlers({
     */
     function callback(): void {
         const pill_widget = get_pill_widget();
-        const pill_user_ids = get_pill_user_ids(pill_widget);
-        action({pill_user_ids});
+        void (async () => {
+            loading.make_indicator($(".add-subscriber-loading-spinner"), {
+                height: 56, // 4em at 14px / 1em
+            });
+            const pill_user_ids = await get_pill_user_ids(pill_widget);
+            loading.destroy_indicator($(".add-subscriber-loading-spinner"));
+            // If we're no longer in the same view after fetching
+            // subscriber data, don't update the UI.
+            if (get_pill_widget() !== pill_widget) {
+                return;
+            }
+            action({pill_user_ids});
```

Comment:
I'm not really sure why we need the loading indicator and then the `if (get_pill_widget() !== pill_widget) {` scenario. Can you help me with an explanation for that? Thanks!

Reply:
`await get_pill_user_ids` can take a while on a slow internet connection, and we want to visibly show something while we fetch data. Once we're done fetching data, the view might have changed (e.g. adding subscribers for a different stream)